engine/new_runtime/src/memories/noncontiguous_memory.rs

```diff
@@ -0,0 +1,372 @@
+use crate::memories::buffer::Buffer;
+use crate::memories::file_memory::FileMemory;
+use crate::crypto_utils::crypto_box::{BoxProvider};
+use crate::locked_memory::ProtectedConfiguration::*;
+use crate::locked_memory::ProtectedMemory;
+use crate::locked_memory::LockedConfiguration::{self, *};
+use crate::locked_memory::LockedMemory;
+use crate::locked_memory::MemoryError::{self, *};
+use crate::types::Bytes;
+use crypto::hashes::sha;
+use core::fmt::{self, Debug, Formatter};
+use zeroize::Zeroize;
+
+
+static IMPOSSIBLE_CASE: &'static str = "NonContiguousMemory: this case should not happen if allocated properly";
+
+// Currently we only support data of 32 bytes in noncontiguous memory
+const NC_DATA_SIZE: usize = 32;
+
+// NONCONTIGUOUS MEMORY
+/// Shards of memory which composes a non contiguous memory
+enum MemoryShard<T: Zeroize + Bytes, P: BoxProvider> {
```

Comment:
Minor: Avoid repeating a suffix in variants, if the enum itself makes the intend clear.